lstchain/image/muon/intensity_fitter.py

```diff
@@ -0,0 +1,562 @@
+    hole_radius_m,
+    impact_parameter_m,
+    phi_rad,
+    center_x_rad,
+    center_y_rad,
+    radius_rad,
+    ring_width_rad,
+    pixel_x_rad,
+    pixel_y_rad,
+    pixel_diameter_rad,
+    oversampling=3,
+    min_lambda_m=300e-9,
+    max_lambda_m=600e-9,
+):
+    """Function for producing the expected image for a given set of trial
+    muon parameters without using astropy units but expecting the input to
+    be in the correct ones.
+    """
+
+    # First produce angular position of each pixel w.r.t muon center
+    dx = pixel_x_rad - center_x_rad
+    dy = pixel_y_rad - center_y_rad
+    ang = np.arctan2(dy, dx)
+    # Add muon rotation angle
+    ang += phi_rad
+
+    # Produce smoothed muon profile
+    ang_prof, profile = create_profile(
+        mirror_radius_m,
+        hole_radius_m,
+        impact_parameter_m,
+        radius_rad,
+        phi_rad,
+        pixel_diameter_rad,
+        oversampling=oversampling,
+    )
+
+    # Produce gaussian weight for each pixel given ring width
+    radial_dist = np.sqrt(dx ** 2 + dy ** 2)
+    # The weight is the integral of the ring's radial gaussian profile inside the
+    # ring's width
+    delta = pixel_diameter_rad / 2
+    cdfs = norm.cdf(
+        [radial_dist + delta, radial_dist - delta], radius_rad, ring_width_rad
+    )
+    gauss = cdfs[0] - cdfs[1]
+
+    # interpolate profile to find prediction for each pixel
+    pred = np.interp(ang, ang_prof, profile)
+
+    # Multiply by integrated emissivity between 300 and 600 nm, and rest of factors to
+    # get total number of photons per pixel
+    # ^ would be per radian, but no need to put it here, would anyway cancel out below
+
+    pred *= alpha * (min_lambda_m ** -1 - max_lambda_m ** -1)
+    pred *= pixel_diameter_rad / radius_rad
+    # multiply by angle (in radians) subtended by pixel width as seen from ring center
+
+    pred *= 0.5*np.sin(2 * radius_rad)
+
+    # multiply by gaussian weight, to account for "fraction of muon ring" which falls
+    # within the pixel
+    pred *= gauss
+
+    # Now it would be the total light in an area S delimited by: two radii of the
+    # ring, tangent to the sides of the pixel in question, and two circles concentric
+    # with the ring, also tangent to the sides of the pixel.
+    # A rough correction, assuming pixel is round, is introduced here:
+    # [pi*(pixel_diameter/2)**2]/ S. Actually, for the large rings (relative to pixel
+    # size) we are concerned with, a good enough approximation is the ratio between a
+    # circle's area and that of the square whose side is equal to the circle's
+    # diameter. In any case, since in the end we do a data-MC comparison of the muon
+    # ring analysis outputs, it is not critical that this value is exact.
+    pred *= CIRCLE_SQUARE_AREA_RATIO
+
+    return pred
+
+
+def calc_likelihood(image, pred, spe_width, ped):
+    """Calculate likelihood of prediction given the measured signal,
+    gaussian approx from [denaurois2009]_
```

Comment:
A more specific biblio would be appreciated. I suppose it's  10.1016/j.astropartphys.2009.09.001 ?